rls/src/main/java/io/grpc/rls/internal/LinkedHashLruCache.java

```diff
@@ -191,6 +191,22 @@ private V invalidate(K key, EvictionType cause) {
     }
   }
 
+  @Override
+  public final boolean invalidate(K key, V value) {
+    checkNotNull(key, "key");
+    checkNotNull(value, "value");
+    synchronized (lock) {
+      SizedValue existing = delegate.get(key);
```

Comment:
`get()` will trigger LinkedHashMap's access ordering. If the existing value is not the one to be deleted, it will be re-inserted to the tail of the linked list. Maybe 
```
boolean replaced = replace(key, value, null);
if (replaced) {
  delegate.remove(key);
  evictionListener.onEviction(key, existing, EvictionType.EXPLICIT);
  return true;
}
```

Reply:
yes, i didn't really consider the access ordering. thanks

Reply:
unfortunately the replace is java8 api. doesn't seem like there is a good option for java7

Reply:
Maybe leave here a TODO with rationale on access ordering and java7 restriction.

Reply:
i am trying to remove this method. and make the caller work around this problem.

Reply:
okay removed.